test/test_main_invocation.py

```diff
@@ -52,9 +52,6 @@ def test_top_level_import(charm_env: None):
 def test_top_level_import_legacy_call(charm_env: None):
     import ops
 
```

Comment:
My original intention was to actually test `ops.main.main(...)` as that remains a supported, albeit deprecated way.

I'd rather see `with...deprecated` removed, but `ops.main.main(CharmBase)` remain.